drivers/serial/uart_nrfx_uarte2.c

```diff
@@ -297,7 +316,6 @@ static void on_rx_buf_req(const struct device *dev)
 	 */
 	if (!K_TIMEOUT_EQ(adata->rx_timeout, K_NO_WAIT)) {
 		/* Read and clear any pending new data information. */
-		nrfx_uarte_rx_new_data_check(nrfx_dev);
```

Comment:
The related comment should be removed together with this call.

Reply:
removed